### DISCUSSION TITLE
Next13 app folder not render html in page source

### DISCUSSION THREAD

**Discussion by F-Moody** [vercel/next.js#46870] [Help]
### Summary

I've been testing Nextjs beta for a while now but i'm noticing that the page-source it's always empty no matter what.

`    

    const Page = () => {
    const headersList = headers();
    const userAgent = headersList.get('user-agent');
    return(
          <div>
            <h1>Test</>
            <ServerComponent/>
              <VisitProvider userAgent={userAgent}>
                <Suspense fallback={<div>Suspending..</div>}>
                  <Visit/>
              </Suspense>
                  <FlightSearcherWrapper/>
              </VisitProvider>
          </div>
      )
    }
  
    export default Page
    
    `
  
Output on the pagesource :

`
    <!DOCTYPE html><html id="__next_error__"><head><link rel="preload" as="script" href="/_next/static/chunks/polyfills.js"/><script src="/_next/static/chunks/polyfills.js" nomodule=""></script></head><body><script src="/_next/static/chunks/webpack.js" async=""></script><script src="/_next/static/chunks/main-app.js" async=""></script></body></html><script>(self.__next_f=self.__next_f||[]).push([0])</script><script>self.__next_f.push([1,"0:\"$L1\"\n"])</script><script>self.__next_f.push([1,"2:I{\"id\":\"(app-client)/./node_modules/next/dist/client/components/app-router.js\",\"name\":\"\",\"chunks\":[\"app-client-internals:app-client-internals\"],\"async\":false}\n4:I{\"id\":\"(app-client)/./node_modules/next/dist/client/components/error-boundary.js\",\"name\":\"\",\"chunks\":[\"app-client-internals:app-client-internals\"],\"async\":false}\n5:I{\"id\":\"(app-client)/./node_modules/next/dist/client/components/layout-router.js\",\"name\":\"\",\"chunks\":[\"app-client-internals:app-client-internals\"],\"async\":false}\n6:I{\"id\":\"(app-client)"])</script><script>self.__next_f.push([1,"/./node_modules/next/dist/client/components/render-from-template-context.js\",\"name\":\"\",\"chunks\":[\"app-client-internals:app-client-internals\"],\"async\":false}\n7:I{\"id\":\"(app-client)/./app/test/(components)/VisitProvider.js\",\"name\":\"\",\"chunks\":[\"app/test/page:app/test/page\"],\"async\":false}\n8:\"$Sreact.suspense\"\n9:I{\"id\":\"(app-client)/./app/test/(components)/Visit.js\",\"name\":\"\",\"chunks\":[\"app/test/page:app/test/page\"],\"async\":false}\na:I{\"id\":\"(app-client)/./app/test/(components)/FlightSearcherWrapper.js\",\"name\":"])</script><script>self.__next_f.push([1,"\"\",\"chunks\":[\"app/test/page:app/test/page\"],\"async\":false}\n"])</script><script>self.__next_f.push([1,"1:[\"$\",\"$L2\",null,{\"assetPrefix\":\"\",\"initialCanonicalUrl\":\"/test\",\"initialTree\":[\"\",{\"children\":[\"test\",{\"children\":[\"\",{}]},null,null,true]}],\"initialHead\":[\"$L3\",[[\"$\",\"title\",null,{}],[\"$\",\"meta\",null,{\"content\":\"width=device-width, initial-scale=1\",\"name\":\"viewport\"}],[\"$\",\"link\",null,{\"rel\":\"icon\",\"href\":\"/favicon.ico\"}]]],\"globalErrorComponent\":\"$4\",\"children\":[\"$\",\"$L5\",null,{\"parallelRouterKey\":\"children\",\"segmentPath\":[\"children\"],\"hasLoading\":false,\"template\":[\"$\",\"$L6\",null,{}],\"childProp\":{\"current\":[[\"$\",\"html\",null,{\"children\":[[\"$\",\"head\",null,{}],[\"$\",\"body\",null,{\"children\":[\"$\",\"$L5\",null,{\"parallelRouterKey\":\"children\",\"segmentPath\":[\"children\",\"test\",\"children\"],\"hasLoading\":false,\"template\":[\"$\",\"$L6\",null,{}],\"notFound\":[[\"$\",\"title\",null,{\"children\":\"404: This page could not be found.\"}],[\"$\",\"div\",null,{\"style\":{\"fontFamily\":\"system-ui,\\\"Segoe UI\\\",Roboto,Helvetica,Arial,sans-serif,\\\"Apple Color Emoji\\\",\\\"Segoe UI Emoji\\\"\",\"height\":\"100vh\",\"textAlign\":\"center\",\"display\":\"flex\",\"flexDirection\":\"column\",\"alignItems\":\"center\",\"justifyContent\":\"center\"},\"children\":[\"$\",\"div\",null,{\"children\":[[\"$\",\"style\",null,{\"dangerouslySetInnerHTML\":{\"__html\":\"body{color:#000;background:#fff;margin:0}.next-error-h1{border-right:1px solid rgba(0,0,0,.3)}@media (prefers-color-scheme:dark){body{color:#fff;background:#000}.next-error-h1{border-right:1px solid rgba(255,255,255,.3)}}\"}}],[\"$\",\"h1\",null,{\"className\":\"next-error-h1\",\"style\":{\"display\":\"inline-block\",\"margin\":\"0 20px 0 0\",\"padding\":\"0 23px 0 0\",\"fontSize\":24,\"fontWeight\":500,\"verticalAlign\":\"top\",\"lineHeight\":\"49px\"},\"children\":\"404\"}],[\"$\",\"div\",null,{\"style\":{\"display\":\"inline-block\"},\"children\":[\"$\",\"h2\",null,{\"style\":{\"fontSize\":14,\"fontWeight\":400,\"lineHeight\":\"49px\",\"margin\":0},\"children\":\"This page could not be found.\"}]}]]}]}]],\"childProp\":{\"current\":[[\"$\",\"div\",null,{\"children\":[[\"$\",\"a\",null,{\"href\":\"/\",\"children\":\"This is a server component\"}],[\"$\",\"$L7\",null,{\"userAgent\":\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36\",\"children\":[[\"$\",\"$8\",null,{\"fallback\":[\"$\",\"div\",null,{\"children\":\"Suspending..\"}],\"children\":[\"$\",\"$L9\",null,{}]}],[\"$\",\"$La\",null,{}]]}]]}],null,null,[]],\"segment\":\"\"}}]}]]}],null,null,[]],\"segment\":\"test\"}}]}]\n"])</script><script>self.__next_f.push([1,"3:[[[\"$\",\"meta\",null,{\"charSet\":\"utf-8\"}],null,null,null,null,null,null,null,null,null,null,[\"$\",\"meta\",null,{\"name\":\"viewport\",\"content\":\"width=device-width, initial-scale=1\"}],null,null,null,null,null,null,null,null,null,null,[]],[null,null,null,null],null,null,[null,null,null,null,null],null,null,null,null,null]\n"])</script>
`

So, not only the server component is just streamed as JSON (i thought should be rendered on the server and sent as html ), but also the Visit component ( which is client and in the documentation it's written that it would be prerendered on the server as well )but most surpisingly the simple h1 is not there in the page source. Can someone clarify this ? I'm concerned about SEO here.

**Comment by joulev** [comment]
It's likely your higher-level layouts hiding the page on prerender, because on a minimal example it does work fine. See e.g. [here](https://github.com/joulev/debug/tree/nextjs-prerender-html-F-Moody), it does have the h1 and everything in the page source (see page source of https://debug-git-nextjs-prerender-html-f-moody-joulev.vercel.app).

**Reply by otterwolf** [reply]
🌜

**Reply by F-Moody** [reply]
Hello @joulev  and thanks for the reply.
Can you clarify what do you mean hiding the page on prerender?

My layout is just the bare minimum :

`export default function RootLayout({ children }) {
  return (
    <html>
      <head />
      <body>{children}</body>
    </html>
  )
}
`

**Reply by joulev** [reply]
@F-Moody It can be anywhere at higher level client components that there is a conditional that hides the page initially but displays it after running something on the browser. Without a reproduction repo from your side though, I can't do much except demonstrating that it is indeed pre-rendered if you set it up correctly.

**Reply by F-Moody** [reply]
@joulev  Ok, i've tried something and it seems that it works. 
It stops working, like in the example provided if i add this Provider on the page : 





`  const VisitProvider = ({children, userAgent}) => {
    const pathname = usePathname()
    
    const [isLoading, setIsLoading] = useState(true)
    const [visit, setVisit] = useState('')
    useEffect( () => {
        async function Fetcher () {
            try {
                const visit = await fetch(".........",{
                    method: 'POST',
                    headers: {
                        'Content-Type': 'application/json'
                    },
                    body: JSON.stringify({
                        website:'something',
                        clientInterface: 'something',
                        userAgent: pathname,
                        entryPoint: userAgent
                    })
                }).then(res => res.json())
                setIsLoading(false)
                setVisit(visit.visitInformation)
            }
            catch (e) {
                console.error(e, 'error')
                setIsLoading(false)
                setVisit("An error as occured")
            }
        }

         Fetcher()
    }, [])

    const memoValue = useMemo(() => ({
        visit,
        isLoading
    }), [visit, isLoading])

    const visitGetter = () => new Promise(resolve => resolve(visit))

    return (
        <VisitContext.Provider value={memoValue}>
            <FrontendApiClientProvider queryClient={....} graphQLClient={....}>
                {children}
            </FrontendApiClientProvider>
        </VisitContext.Provider>
    )
}

export default VisitProvider`

Can you help me understand why? 

Basically if my Page is this : 

`const Page = () => {
  const headersList = headers();
  const userAgent = headersList.get('user-agent');
  return(
        <div>
          <ServerComponent/>
            <VisitProvider userAgent={userAgent}/>
        </div>
    )
}`

It will not work, it will not send the ServerComponent as html 

But if i remove the VisitProvider component it will.

Why?
How does the VisitProvider affects the rest of the page?

**Reply by joulev** [reply]
Without a minimal reproduction repo I can't say much, as I said. Content of individual files is not too helpful.
